pkg/env/readers.go

```diff
@@ -71,16 +112,11 @@ func testReader(fallback cfg.Reader, overrides map[string]interface{}) cfg.Reade
 //
 // The provided value is serialized to yaml and so can be structured data.
 func FakeTestConfig(fName string, ptr interface{}) func() {
-	if _, ok := testOverrides[fName]; ok {
-		// This is not ideal.  We would prefer to return an error.
-		// However, this function's signature does not support it and we
-		// don't want to incur the backwards-incompatibility of changing
-		// it right now.
-		panic(fmt.Errorf("repeated test override of '%s'", fName))
-	}
+	// add ensures that it doesn't already exist to prevent two tests running
```

Comment:
Document panic in function signature, and document potential leakage to other tests?

This still leaks shared state: You can read overrides of _other_ functions, even if you're only mocking out one.

This whole thing is a nightmare, and honestly the right call might be to make a breaking change (or create a correct version clients could migrate to).

At the bare minimum this needs a TODO + Jira bug.

Reply:
Okay, I'm going to create a jira bug because I was just kinda doing this one out of kindness - I was originally just going to open a DT bug but decided to try and help out. I can't really spend anymore time on this with other commitments right now - this is kind of a tight sprint w/the holidays.

Reply:
https://outreach-io.atlassian.net/browse/DT-3185